### PR TITLE
Rename all gdnative occurences to gdextension

### PR DESCRIPTION
**Pull request by groud**
I guess it's kind of needed before 4.0.

It at least compiles on my computer, so I don't think there are any rename that caused an issue for now. I hope I did not rename too much, though I checked most changes (translations and 3 to 4 converter renames were removed for example)

It will need a correspondind godot-cpp update though.

Non-exhaustive list of case-sensitive renames:

GDExtension -> GDNative
GDNATIVE -> GDEXTENSION
gdextension -> gdnative
ExtensionExtension ->Extension (for where there was GDNativeExtension)
EXTENSION_EXTENSION ->EXTENSION (for where there was GDNATIVE_EXTENSION)
gdnlib -> gdextension
gdn_interface -> gde_interface
gdni -> gde_interface

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs [hunk 1]
```diff
@@ -153,9 +153,9 @@ public static partial void
         internal static partial void godotsharp_callable_call_deferred(in godot_callable p_callable,
             godot_variant** p_args, int p_arg_count);
 
-        // GDNative functions
+        // GDExtension functions
 
-        // gdnative.h
+        // gdextension.h
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs:158]: This doesn't make much sense, as this mimics functions from the old the old `gdnarive.h` header. Also, AFAIK `gdextension.h` doesn't exist either.

File: modules/mono/interop_types.h [hunk 2]
```diff
@@ -40,7 +40,7 @@ extern "C" {
 #include <stdbool.h>
 #include <stdint.h>
 
-// This is taken from the old GDNative, which was removed.
+// This is taken from the old GDExtension, which was removed.
```
Comment (neikeq) [inline modules/mono/interop_types.h:43]: The comment says this was taken from the old GDNative headers, which were removed. Renaming this here would cause confusion.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs (hunk 1 above)
Comment (groud) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs:158]: Hmm, well, I renamed `native_extension.h` to `gdextension.h` now, so that file exists. But indeed it does not correspond to the old `gdnative.h` file. So I agree on the change, I'll fix it.

File: platform/android/java/lib/src/org/godotengine/godot/plugin/GodotPlugin.java [hunk 3]
```diff
@@ -71,9 +71,9 @@
...
  * extending {@link GodotPlugin}.
  *
- * A plugin can also define and provide c/c++ gdnative libraries and nativescripts for the target
+ * A plugin can also define and provide c/c++ gdextension libraries and nativescripts for the target
  * app/game to leverage.
- * The shared library for the gdnative library will be automatically bundled by the aar build
+ * The shared library for the gdextension library will be automatically bundled by the aar build
  * system.
  * Godot '*.gdnlib' and '*.gdns' resource files must however be manually defined in the project
  * 'assets' directory. The recommended path for these resources in the 'assets' directory should be:
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/plugin/GodotPlugin.java:79]: To be thorough, the comments will need to be updated as well to match the gdextension workflow (e.g: replace occurrences of `gdnlib` and `gdns` with `gdextension`).

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -118,13 +118,13 @@ doc_classes/*                       @godotengine/documentation
 
 ## Text
 /modules/freetype/                  @godotengine/buildsystem
-/modules/gdnative/text/             @godotengine/gui-nodes
+/modules/gdextension/text/          @godotengine/gui-nodes
```
Comment (akien-mga) [inline .github/CODEOWNERS:121]: These don't exist, should just be removed.

File: core/extension/gdextension_interface.cpp [hunk 5]
```diff
@@ -0,0 +1,1083 @@
...
+}
+
+static void *gdextension_classdb_get_class_tag(GDExtensionConstStringNamePtr p_classname) {
+	const StringName classname = *reinterpret_cast<const StringName *>(p_classname);
+	ClassDB::ClassInfo *class_info = ClassDB::classes.getptr(classname);
+	return class_info ? class_info->class_ptr : nullptr;
+}
+
+void gdextension_setup_interface(GDExtensionInterface *p_interface) {
+	GDExtensionInterface &gdni = *p_interface;
```
Comment (akien-mga) [inline core/extension/gdextension_interface.cpp:913]: Nitpick but technically this identifier should be `gdei`.

File: core/variant/variant_internal.h [hunk 6]
```diff
@@ -1517,7 +1517,7 @@ struct VariantTypeAdjust<Object *> {
 	}
 };
 
-// GDNative extension helpers.
+// GDExtension extension helpers.
```
Comment (akien-mga) [inline core/variant/variant_internal.h:1520]: Can remove the " extension " part I guess.

File: main/main.cpp [hunk 7]
```diff
@@ -423,7 +423,7 @@ void Main::print_help(const char *p_binary) {
 	OS::get_singleton()->print("  --doctool [<path>]                Dump the engine API reference to the given <path> (defaults to current dir) in XML format, merging if existing files are found.\n");
 	OS::get_singleton()->print("  --no-docbase                      Disallow dumping the base types (used with --doctool).\n");
 	OS::get_singleton()->print("  --build-solutions                 Build the scripting solutions (e.g. for C# projects). Implies --editor and requires a valid project to edit.\n");
-	OS::get_singleton()->print("  --dump-gdextension-interface      Generate GDExtension header file 'gdnative_interface.h' in the current folder. This file is the base file required to implement a GDExtension.\n");
+	OS::get_singleton()->print("  --dump-gdextension-interface      Generate GDExtension header file 'gdextension_interface.h' in the current folder. This file is the base file required to implement a GDExtension.\n");
 	OS::get_singleton()->print("  --dump-extension-api              Generate JSON dump of the Godot API for GDExtension bindings named 'extension_api.json' in the current folder.\n");
```
Comment (akien-mga) [inline main/main.cpp:427]: I wonder if we should rename this one to `--dump-gdextension-api` to match `--dump-gdextension-interface`... but then should the JSON be `gdextension_api.json` :thinking:

:canned_food: :worm: 

Scope creep anyway for this PR ;)

File: main/main.cpp [hunk 8]
```diff
@@ -1063,8 +1063,8 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 			// Register as an editor instance to use low-end fallback if relevant.
 			editor = true;
 			cmdline_tool = true;
-			dump_gdnative_interface = true;
-			print_line("Dumping gdnative interface header file");
+			dump_gdextension_interface = true;
+			print_line("Dumping gdextension interface header file");
```
Comment (akien-mga) [inline main/main.cpp:1067]: We could capitalize "GDExtension".

File: modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs [hunk 9]
```diff
@@ -153,7 +153,7 @@ public static partial void
         internal static partial void godotsharp_callable_call_deferred(in godot_callable p_callable,
             godot_variant** p_args, int p_arg_count);
 
-        // GDNative functions
+        // GDExtension functions
 
         // gdnative.h
```
Comment (akien-mga) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs:158]: This probably warrants further cleanup by @godotengine/dotnet - for this PR I'd suggest leaving it unchanged (same with `modules/mono/glue/runtime_interop.cpp`) so it stays clear that it's an outdated comment and that this API requires a review.

**Review by akien-mga — APPROVED** [review]
Looks good!

Left a couple nitpicks but overall this seems ready to merge.

File: core/extension/gdextension_interface.cpp (hunk 5 above)
Comment (akien-mga) [reply core/extension/gdextension_interface.cpp:913]: Or `gde_interface` might be more readable, as we use in godot-cpp.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs (hunk 9 above)
Comment (groud) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/NativeFuncs.cs]: OK that make sense, I reverted the rename in comments there.

File: main/main.cpp (hunk 7 above)
Comment (groud) [reply main/main.cpp:427]: You tell me. But I think I agree with the fact it could go in another PR :grin: .
I think we could discuss that in the GDExtension meeting, I'll add it as a topic to discuss.

**Comment by akien-mga** [comment]
Thanks!
